examples/EvaluatingChatAssistant/EvaluatingChatAssistant.ipynb

```diff
@@ -0,0 +1,353 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import { Eval } from \"braintrust\";\n",
+    "import dataset from \"./assets/dataset\"\n",
+    "import factualitySpec from \"./assets/factuality\";\n",
+    "import { LLMClassifierFromSpec } from \"autoevals\";\n",
+    "\n",
+    "function Factuality(args: {\n",
+    "  input: {\n",
+    "    input: string,\n",
+    "    chat_history: {\n",
+    "      role: \"user\" | \"assistant\";\n",
+    "      content: string;\n",
+    "    }[]\n",
+    "  },\n",
+    "  output: string,\n",
+    "  expected: string\n",
+    "}) {\n",
+    "  const scorer = LLMClassifierFromSpec(\n",
+    "    \"Factuality\",\n",
+    "    {\n",
+    "      ...factualitySpec,\n",
+    "      model: \"gpt-4o\"\n",
+    "    }\n",
+    "  );\n",
+    "  return scorer(args);\n",
+    "}\n",
+    "\n",
+    "Eval(\"Chat assistant\", {\n",
+    "  experimentName: \"gpt-4o assistant\",\n",
+    "  data: () => dataset,\n",
+    "  task: runTask,\n",
+    "  scores: [Factuality],\n",
+    "  metadata: {\n",
+    "    model: \"gpt-4o\",\n",
+    "    prompt: \"You are a helpful, polite assistant who knows about sports.\"\n",
+    "  }\n",
+    "});"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "Experiment gpt - 4o assistant is running at http://www.braintrust.dev/app/braintrustdata.com/p/Chat%20assistant/experiments/gpt-4o%20assistant\n",
+    " ████████████████████████████████████████ | Chat assistant[experimentName = gpt - 4o... | 100 % | 5 / 5 datapoints\n",
+    "\n",
+    "=========================SUMMARY=========================\n",
+    "64.00% 'Factuality' score       (0 improvements, 0 regressions)\n",
+    "\n",
+    "4.28s 'duration'        (0 improvements, 0 regressions)\n",
+    "\n",
+    "See results for gpt-4o assistant at http://www.braintrust.dev/app/braintrustdata.com/p/Chat%20assistant/experiments/gpt-4o%20assistant"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Visualizing\n",
+    "\n",
+    "Now that we've run the eval, we can look at the results on [braintrust.dev](https://www.braintrust.dev).\n",
+    "\n",
+    "Open the newly created `Chat assistant` project and you can see the experiment we just ran:\n",
+    "\n",
+    "![dist-chart](./assets/assistant-dist.png)\n",
+    "\n",
+    "The model scored 64% on Factuality, but we can see from the chart that the model responses got a lot of partial credit (as opposed to a few perfect responses and a few completely incorrect responses). The responses were either exactly the same as the expected answer or a subset/superset but still fully factually consistent with the expected answer."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Comparing to a model with no context\n",
```

Comment:
Would you be able to structure the cookbook so that the lower score eval runs first? Then we can show improvements by adding the chat history

e.g. `Let's see how the prompts perform when no chat history is provided whatever whatever`

Reply:
Also this makes me think if you were able to create another data set with even more specific chat history that would perform even better, and we could compare the 64% to that one as another iteration.

I think my first reaction to seeing 64% as a final result is that it's kind of low, so it would be nice to have a final result that was higher